### PR TITLE
Add methods to make a Variant locatable

### PR DESCRIPTION
**Pull request by nh13**
Alternate implementation of https://github.com/fulcrumgenomics/fgbio/pull/696

This gives two ways:
- use `toLocatable` on `Variant` directly
- import the implicit class `VariantLocatable`

The `VariantLocatable` class is just a container class for `Variant`, implementing the required methods for `Locatable`.  So you can use `VariantLocatable` anywhere `Locatable` is required, and still return the variant.

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala [hunk 1]
```diff
@@ -361,4 +360,13 @@ private[api] object VcfConversions {
 
     out
   }
+
+  /**
+    * Little container class for a [[Variant]] to make it locatable (see [[Locatable]]).
+    */
+  implicit case class VariantLocatable(variant: Variant) extends Locatable {
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala:367]:
```suggestion
  implicit class VariantLocatable(variant: Variant) extends Locatable {
```

File: src/main/scala/com/fulcrumgenomics/vcf/api/Variant.scala [hunk 2]
```diff
@@ -24,6 +24,8 @@
 
 package com.fulcrumgenomics.vcf.api
 
+import com.fulcrumgenomics.vcf.api.VcfConversions.VariantLocatable
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/vcf/api/Variant.scala:27]: This doesn't belong here.  Note that `VcfConvefsions` is `private[api]` and is documented as follows:

```
  * Object that provides methods for converting from fgbio's scala VCF classes to HTSJDK's
  * Java VCF-related classes and vice-versa.  Only intended for internal use within the `api`
  * package and should not be exposed publicly.
```

File: src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala [hunk 3]
```diff
@@ -361,4 +360,13 @@ private[api] object VcfConversions {
 
     out
   }
+
+  /**
+    * Little container class for a [[Variant]] to make it locatable (see [[Locatable]]).
+    */
+  implicit class VariantLocatable(variant: Variant) extends Locatable {
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala:367]: Wouldn't it make more sense to call this `LocatableVariant`?

**Review by tfenne — CHANGES_REQUESTED** [review]

**Comment by nh13** [comment]
@tfenne one more look?

File: src/main/scala/com/fulcrumgenomics/vcf/api/Variant.scala [hunk 4]
```diff
@@ -69,6 +70,15 @@ object Variant {
   private[api] val EmptyInfo:       ListMap[String, Any]  = ListMap.empty
   private[api] val EmptyGtAttrs:    Map[String, Any]      = Map.empty
   private[api] val EmptyGenotypes:  Map[String, Genotype] = Map.empty
+
+  /**
+    * Little container class for a [[Variant]] to make it locatable (see [[Locatable]]).
+    */
+  implicit case class LocatableVariant(variant: Variant) extends Locatable {
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/vcf/api/Variant.scala:77]:
```suggestion
  implicit class LocatableVariant(variant: Variant) extends Locatable {
```

File: src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala [hunk 5]
```diff
@@ -24,19 +24,17 @@
...
-
 import com.fulcrumgenomics.FgBioDef._
 import com.fulcrumgenomics.fasta.SequenceMetadata
 import com.fulcrumgenomics.vcf.api.Allele.NoCallAllele
 import com.fulcrumgenomics.vcf.api.VcfCount.Fixed
 import htsjdk.variant.variantcontext.{GenotypeBuilder, VariantContext, VariantContextBuilder, Allele => JavaAllele}
 import htsjdk.variant.vcf._
 
+import java.util
+import java.util.{List => JavaList}
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala:35]: Can these go away again now?

**Review by tfenne — APPROVED** [review]

File: src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala (hunk 5 above)
Comment (nh13) [reply src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala:35]: they just got moved, keeping them for now so I can merge the PR

**Pull request merged**